tests/rptest/tests/acls_test.py

```diff
@@ -321,6 +321,101 @@ def test_mtls_principal(self, rules=None, fail=False):
             except ClusterAuthorizationError:
                 assert fail, "list acls should have succeeded"
 
+    @cluster(num_nodes=3)
+    @parametrize(authn_method='sasl')
+    @parametrize(authn_method='mtls_identity')
+    def test_security_feature_migration(self, authn_method: str):
+        def check_perms():
+            try:
+                self.get_client("base").acl_list()
+                self.logger.debug('list acls should have failed for base user')
+                return False
+            except ClusterAuthorizationError:
+                pass
+
+            # should succeed
+            try:
+                self.get_client("cluster_describe").acl_list()
+            except Exception:
+                self.logger.debug(
+                    'list acls should have passed for cluster user')
+                return False
+
+            try:
+                self.get_super_client().acl_list()
+            except Exception:
+                self.logger.debug(
+                    'list acls should have passed for super user')
+                return False
+
+            return True
+
+        self.prepare_cluster(use_tls=True, use_sasl=True)
+
+        wait_until(check_perms,
+                   timeout_sec=90,
+                   err_msg='check perms failed before migration')
+
+        # Change the authn method and initiate feature migration
+        # with a rolling restart. Do not recreate tls certs.
+        self.security.endpoint_authn_method = authn_method
+        self.redpanda.set_security_settings(self.security, create_certs=False)
+        self.logger.debug('Initiating rolling restart')
+        self.redpanda.rolling_restart_nodes(self.redpanda.nodes,
+                                            start_timeout=90,
+                                            stop_timeout=90)
+
+        # Check that the authn_method was set on all brokers
+        pattern = 'Started Kafka API server.*:{' + authn_method + '}.*'
+        assert self.redpanda.search_log(pattern)
+
+        # Once restart is complete, check permissions should succeed when authn_method
+        # is sasl because the system will validate against SASL/SCRAM creds already stored
+        # on the brokers. However, when authn_method is mtls_identiy, check permissions
+        # should fail because no principal mapping rule is set yet.
+        msg = 'check perms failed in intermediate state'
+        if authn_method == 'sasl':
+            wait_until(check_perms, timeout_sec=90, err_msg=msg)
+        elif authn_method == 'mtls_identity':
+            try:
+                wait_until(check_perms, timeout_sec=90, err_msg=msg)
+                raise RuntimeError(
+                    'check perms should have failed in intermediate state')
+            except TimeoutError:
+                # Expect a timeout failure
```

Comment:
Can you explain why?

Is it possible to make a positive check that takes less time?

Reply:
> Can you explain why?
The reason was in the comment
`# However, when authn_method is mtls_identiy, check permissions should fail because no principal mapping rule is set yet.`


> Is it possible to make a positive check that takes less time?

I'll decrease the timeout_sec here instead because there's no need to retry check perms for that long. I don't see how a positive check would change the timeouts here since I'm testing for a failure but I could just be tired.